<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>835</width>
    <height>649</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>500</width>
      <height>581</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideNone</enum>
    </property>
    <property name="usesScrollButtons">
     <bool>true</bool>
    </property>
    <property name="documentMode">
     <bool>false</bool>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <property name="movable">
     <bool>false</bool>
    </property>
    <property name="tabBarAutoHide">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="w_data">
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <attribute name="title">
      <string>data</string>
     </attribute>
     <widget class="QTextEdit" name="t_output">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>494</width>
        <height>551</height>
       </rect>
      </property>
      <property name="documentTitle">
       <string/>
      </property>
      <property name="acceptRichText">
       <bool>false</bool>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="w_graph">
     <attribute name="title">
      <string>graph</string>
     </attribute>
     <widget class="MyChart" name="g_chart" native="true">
      <property name="geometry">
       <rect>
        <x>-1</x>
        <y>-1</y>
        <width>501</width>
        <height>551</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="b_send">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>530</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>send</string>
    </property>
   </widget>
   <widget class="QComboBox" name="c_portchoose">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>70</y>
      <width>161</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>null</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="c_ratechoose">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>130</y>
      <width>87</width>
      <height>22</height>
     </rect>
    </property>
    <property name="currentText">
     <string>1200</string>
    </property>
    <item>
     <property name="text">
      <string>1200</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2400</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4800</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>9600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>19200</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>38400</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>57600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>115200</string>
     </property>
    </item>
   </widget>
   <widget class="QTextEdit" name="t_send">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>370</y>
      <width>211</width>
      <height>141</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="l_port">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>40</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>port：</string>
    </property>
   </widget>
   <widget class="QLabel" name="l_rate">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>100</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Baud Rate：</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>170</y>
      <width>121</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>move cursor</string>
    </property>
   </widget>
   <widget class="QToolButton" name="bt_more">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>130</y>
      <width>47</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>290</y>
      <width>101</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>clear data</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>250</y>
      <width>101</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>clear graph</string>
    </property>
   </widget>
   <widget class="QPushButton" name="b_fresh">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>70</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>fresh</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>835</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufiel">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="a_open_file"/>
    <addaction name="a_save_file"/>
    <addaction name="a_saveas"/>
    <addaction name="a_close_file"/>
    <addaction name="a_export_pic"/>
    <addaction name="a_exit"/>
   </widget>
   <widget class="QMenu" name="menuabout">
    <property name="title">
     <string>help</string>
    </property>
    <addaction name="actionhelp"/>
    <addaction name="actionabout"/>
   </widget>
   <addaction name="menufiel"/>
   <addaction name="menuabout"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="a_open_file">
   <property name="text">
    <string>open file</string>
   </property>
  </action>
  <action name="a_save_file">
   <property name="text">
    <string>save file</string>
   </property>
  </action>
  <action name="a_close_file">
   <property name="text">
    <string>close file</string>
   </property>
  </action>
  <action name="a_exit">
   <property name="text">
    <string>exit</string>
   </property>
  </action>
  <action name="a_export_pic">
   <property name="text">
    <string>export pic</string>
   </property>
  </action>
  <action name="actionhelp">
   <property name="text">
    <string>help</string>
   </property>
  </action>
  <action name="actionabout">
   <property name="text">
    <string>about</string>
   </property>
  </action>
  <action name="a_saveas">
   <property name="text">
    <string>save as</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyChart</class>
   <extends>QWidget</extends>
   <header>mychart.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>a_open_file</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a_save_file</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>savefile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a_export_pic</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>savepic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a_saveas</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save_asfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openfile()</slot>
  <slot>savefile()</slot>
  <slot>save_asfile()</slot>
  <slot>savepic()</slot>
 </slots>
</ui>
